pkg/private/install.py.tpl

```diff
@@ -103,15 +103,27 @@ class NativeInstaller(object):
 
     def _install_treeartifact(self, entry):
         logging.info("COPYTREE %s <- %s/**", entry.dest, entry.src)
-        raise NotImplementedError("treeartifact installation not yet supported")
-        for root, dirs, files in os.walk(entry.src):
-            relative_installdir = os.path.join(entry.dest, root)
+        shutil.copytree(
+            src = entry.src,
+            dst = entry.dest,
+            symlinks = True,
+            copy_function=lambda s, d: self._do_file_copy(s, d, entry.mode, entry.user, entry.group),
```

Comment:
Is entry.mode the right thing? That would be the top level dir of the artifact, not a typical file within it.

Reply:
I would assume that `entry` is for the WHOLE tree artifact, so -- for all directories & files within it. I can check what pkg_tar/pkg_zip does and try aligning the behavior too.

Reply:
As far as I can see, the "intermediate directories" within the tree artifact is just set with a hard-coded 755:

https://github.com/bazelbuild/rules_pkg/blob/main/pkg/private/tar/build_tar.py#L285

So I did the same thing here.

For the files within the TreeArtifact, I did use entry.mode. But I haven't double check that's the same behavior in pkg_tar/pkg_zip, so I'll leave this open until I confirm this.

Reply:
I have checked that pkg_tar uses the same behavior here (that is, the files inside the TreeArtifact has the same attribute as the TreeArtifact's `attributes` as well). As such, I think it is doing the right thing now, and I have also added tests to mandate this behavior.